model-api-gen/src/main/kotlin/org/modelix/metamodel/generator/ReadonlyGeneratorInput.kt

```diff
@@ -0,0 +1,61 @@
+package org.modelix.metamodel.generator
+
+import org.modelix.model.data.PropertyType
+
+@RequiresOptIn(message = "This API is experimental. It may be changed in the future without notice.")
+@Retention(AnnotationRetention.BINARY)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
+annotation class UnstableGeneratorFeature
+interface ReadonlyProcessedLanguageSet {
+    fun getLanguages(): List<ReadonlyProcessedLanguage>
+}
+
+interface ReadonlyProcessedLanguage {
+    val name: String
+    val languageSet: ReadonlyProcessedLanguageSet
+    fun getConcepts(): List<ReadonlyProcessedConcept>
+}
+
+interface ReadonlyProcessedConcept : ReadonlyProcessedDeprecatable {
+    val name: String
+    val uid: String?
+    val language: ReadonlyProcessedLanguage
+    fun fqName(): String
+
+    fun getDirectSuperConcepts(): Sequence<ReadonlyProcessedConcept>
+    fun getAllSuperConceptsAndSelf(): Sequence<ReadonlyProcessedConcept>
+    fun getDuplicateSuperConcepts(): List<ReadonlyProcessedConcept>
+
+    fun getOwnRoles(): List<ReadonlyProcessedRole>
+}
+
+sealed interface ReadonlyProcessedRole : ReadonlyProcessedDeprecatable {
```

Comment:
I need to take a look at theclass  hierarchy of `ReadonlyProcessedRole`. 
Also, I should look in general over the other hierarchies.